e2e/cypress/integration/mark_as_unread/unread_toast_count_spec.js

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {markAsUnreadByPostIdFromMenu} from './helpers';
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. # Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+// Group: @mark_as_unread
+
+describe('Verify unread toast appears after repeated manual marking post as unread', () => {
+    beforeEach(() => {
+        cy.apiAdminLogin();
+        cy.apiInitSetup().then(({team, channel, user}) => {
+            cy.apiCreateUser().then(({user: user2}) => {
+                cy.apiAddUserToTeam(team.id, user2.id).then(() => {
+                    cy.apiAddUserToChannel(channel.id, user2.id);
+
+                    // # Create a page full of messages
+                    Cypress._.times(30, (i) => {
+                        cy.postMessageAs({
+                            sender: user2,
+                            message: `post${i}`,
+                            channelId: channel.id,
+                        });
+                    });
+
+                    // # Make a mentioned message
+                    cy.postMessageAs({
+                        sender: user2,
+                        message: `hi @${user.username}`,
+                        channelId: channel.id,
+
+                    });
+
+                    // # Login as user
+                    cy.apiLogin(user);
+
+                    // # Visit the channel
+                    cy.visit(`/${team.name}/channels/${channel.name}`);
+
+                    // # Post a reply on RHS
+                    cy.getLastPostId().then((postId) => {
+                        cy.clickPostCommentIcon(postId);
+                        cy.get('#rhsContainer').should('be.visible');
+                        const replyMessage = 'A reply to an older post';
+                        cy.postMessageReplyInRHS(replyMessage);
+                    });
+                });
+            });
+        });
+    });
+    it('MM-T254 Rehydrate mention badge after post is marked as Unread', () => {
+        // # Mark the first post as unread
+        cy.getNthPostId(1).then((postId) => {
+            markAsUnreadByPostIdFromMenu(postId);
+        });
+
+        // * Check that the toast is visible
+        cy.get('div.toast').should('be.visible').then(() => {
+            // * Check that the message is correct and all the mentions and replies are counted in toast
+            cy.get('div.toast__message>span').should('be.visible').first().contains('32 new messages');
+        });
```

Comment:
The test is missing the following assertion:
`In particular, at-here should mention the user even if they were offline when the post was originally made.`

Suggested changes:
```
        cy.get(`#sidebarItem_${currentChannel.name}`).
            scrollIntoView().
            find('#unreadMentions').
            should('be.visible').
            and('have.text', '1');
```

Note: You need to declare and assign the variable currentChannel before.